timely/src/progress/subgraph.rs

```diff
@@ -785,3 +780,52 @@ impl<T: Timestamp> Drop for PerOperatorState<T> {
         self.shut_down();
     }
 }
+
+/// An dummy operator to act as a represenantive of a scope
+struct EmptyOperator<T: Timestamp> {
+    inputs: usize,
+    outputs: usize,
+    shared_progress: SharedProgress<T>,
+}
+
+impl<T: Timestamp> EmptyOperator<T> {
+    fn new(inputs: usize, outputs: usize) -> Self {
+        EmptyOperator {
+            inputs,
+            outputs,
+            shared_progress: SharedProgress::new(inputs, outputs),
+        }
+    }
+}
+
+impl<T: Timestamp> Operate<T> for EmptyOperator<T> {
+    fn local(&self) -> bool {
+        false
+    }
+    fn inputs(&self)  -> usize {
+        self.inputs
+    }
+    fn outputs(&self) -> usize {
+        self.outputs
+    }
+    fn get_internal_summary(&mut self) -> Vec<Vec<Antichain<T::Summary>>> {
+        Vec::new()
+    }
```

Comment:
I wonder (I can try it) whether this should perhaps be a `panic!` in order to avoid miscommunicating. Afaiu, this struct exists only to house the `SharedProgress<T>` struct, and doing anything else with it seems scary.

Reply:
Good point, I added an `unreachable!()` call with an appropriate message